sequential/seq2pat.py

```diff
@@ -283,9 +293,27 @@ class Seq2Pat:
         sequences but no constraints are used to run the mining efficiently and practically.
         Power users can choose to drop this constraint by setting it to be None or increase the maximum span
         as the system has resources to support.
+    batch_size: Optional[int]
```

Comment:
What's the relationship between batch_size and n_jobs. Do you split data into S sizes, that gives B bactches, and we execute on B batches with N jobs? 

Also, is this gonna be reproducible?

Reply:
Need more commentary on how batching works. Is this batched sequentially, or randomly etc. and how is the results aggregated

Reply:
Yes, the entire set is split into B batches with each batch having `batch_size` sequences. Then I map these batches to N jobs.

Before the entire set is split, I shuffle the sequences, trying to make the sequences uniformly distributed. Then batches are split sequentially. There is a random seed used for data shuffle, to make it reproducible.

The patterns are aggregated from mining the batches by calculating the sum of their occurrences.

Reply:
The `min_frequency_df` parameter is used to reduce min_frequency threshold for each batch. It should be set small enough to capture the patterns those should satisfy the min_frequency threshold while running on the entire set. 

Theoretically there is a chance that the batching results will be different from non-batching results. But a small `min_frequency_df` parameter will make the chance to be minimal and thus we have the same results as running on entire set.

Reply:
Just trying to understand.. If `min_frequency` was `4` and we set batch size to be half the whole dataset, couldn't we use `min_frequency` of `2` for each batch to get a similar result?

Reply:
I am trying to understand this, but still confused. In particular, not sure why we need both of batch_size and n_jobs together. And on top, we also need min_freq_df, which I am not sure either. 

I have a feeling that we should be able to simplify this

Reply:
> Just trying to understand.. If `min_frequency` was `4` and we set batch size to be half the whole dataset, couldn't we use `min_frequency` of `2` for each batch to get a similar result?

Yes, that is the main idea for batching process to work. Reducing `min_frequency` (as a percentage) is the way used to apply a lower threshold for each batch. However, for some patterns that can pass this threshold on entire set, those cannot pass the new threshold on each batch. We need to figure out the way to apply an even lower threshold for each batch.

Reply:
> I am trying to understand this, but still confused. In particular, not sure why we need both of batch_size and n_jobs together. And on top, we also need min_freq_df, which I am not sure either.
> 
> I have a feeling that we should be able to simplify this

For using `batch_size` and `n_jobs`, the idea is to control the data size loaded in for one mining task, so this will use less memory for each task, and leave it to users to decide how many cpus can be used. 

For `min_freq_df` parameter, it defines a degree that we want to reduce the threshold for each batch, since not all patterns that should be included in the results as it is done on entire set, would pass the original threshold on a batch, a subset of data.